test/eos_unit_tests.cpp

```diff
@@ -187,6 +187,59 @@ SCENARIO("Relativistic EOS", "[EOSBuilder][RelativisticEOS]") {
   }
 }
 
+SCENARIO("EOS Unit System", "[EOSBuilder][UnitSystem]") {
+  GIVEN("Parameters for an ideal gas") {
+    constexpr Real Cv = 2.0;
+    constexpr Real gm1 = 0.5;
+    EOSBuilder::EOSType type = EOSBuilder::EOSType::IdealGas;
+    EOSBuilder::modifiers_t modifiers;
+    EOSBuilder::params_t base_params, units_params;
+    base_params["Cv"].emplace<Real>(Cv);
+    base_params["gm1"].emplace<Real>(gm1);
+    GIVEN("Units with a thermal unit system") {
+      constexpr Real rho_unit = 1e1;
+      constexpr Real sie_unit = 1e-1;
+      constexpr Real temp_unit = 1;
```

Comment:
`1` probably isn't a great value for testing unit conversions (same for next test case), I would actually opt for oddball numbers with a few digits after the decimal point, small point though

Reply:
:+1: